common/src/http.rs

```diff
@@ -1,9 +1,156 @@
-use crate::errors::HttpError;
-use ic_cdk::api::management_canister::http_request::HttpResponse;
-
-pub async fn http_get(url: &str) -> Result<HttpResponse, HttpError> {
-    let req = ic_http::create_request().get(url).build();
-    // TODO: should we pass cycles in http_get method or we should have a default one?
-    let resp = ic_http::http_request(req, 2_603_101_200).await?;
-    Ok(resp.0)
+// Values are taken from https://internetcomputer.org/docs/current/developer-docs/gas-cost
+// for a 13-node app subnet.
+pub const HTTP_OCALL_BASE_COST: u128 = 49_140_000;
+pub const HTTP_OCALL_REQ_BYTE_COST: u128 = 5_200;
+pub const HTTP_OCALL_RESP_BYTE_COST: u128 = 10_400;
+
+// TODO: should we pass cycles in http_x methods or we should have a default one?
+pub const DEFAULT_HTTP_OUTCALL_COST: u128 = HTTP_OCALL_BASE_COST
+    + 5 * 1024 * HTTP_OCALL_REQ_BYTE_COST
+    + 5 * 1024 * HTTP_OCALL_RESP_BYTE_COST;
+
+pub struct HttpResponse {
+    pub status: u16,
+    pub body: Vec<u8>,
+}
+
+#[cfg(target_arch = "wasm32")]
+pub use icp::{get, post};
+
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::{get, post};
+
+#[cfg(target_arch = "wasm32")]
+mod icp {
+    use ic_cdk::api::{
+        call::CallResult,
+        management_canister::http_request::{
+            CanisterHttpRequestArgument, HttpHeader, HttpMethod,
+            HttpResponse as CanisterHttpResponse,
+        },
+    };
+
+    use crate::errors::HttpError;
+
+    use super::{HttpResponse, DEFAULT_HTTP_OUTCALL_COST};
+
+    pub async fn get(url: &str) -> Result<HttpResponse, HttpError> {
+        let req = CanisterHttpRequestArgument {
+            url: url.to_owned(),
+            method: HttpMethod::GET,
+            ..Default::default()
+        };
+        let resp = ic_http_outcall(req, DEFAULT_HTTP_OUTCALL_COST).await?;
+        resp.0.try_into()
+    }
+
+    pub async fn post(
+        url: &str,
+        headers: &[(&str, &str)],
+        body: Vec<u8>,
+    ) -> Result<HttpResponse, HttpError> {
+        let headers = headers
+            .iter()
+            .map(|(name, value)| HttpHeader {
+                name: name.to_string(),
+                value: value.to_string(),
+            })
+            .collect();
+        let req = CanisterHttpRequestArgument {
+            url: url.to_string(),
+            method: HttpMethod::POST,
+            headers,
+            body: Some(body),
+            ..Default::default()
+        };
+        let resp = ic_http_outcall(req, DEFAULT_HTTP_OUTCALL_COST).await?;
+        resp.0.try_into()
+    }
+
+    async fn ic_http_outcall(
```

Comment:
remove this and replace it with [`http_request`](https://docs.rs/ic-cdk/latest/ic_cdk/api/management_canister/http_request/fn.http_request.html)

Reply:
done